ttnn/cpp/ttnn/operations/experimental/argmax/argmax.hpp

```diff
@@ -0,0 +1,50 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "ttnn/run_operation.hpp"
+#include "ttnn/decorators.hpp"
+#include "ttnn/operations/core/core.hpp"
+
+namespace ttnn {
+namespace operations::experimental {
+
+Tensor _argmax(const Tensor& input_t, int64_t _dim, bool all, const std::optional<MemoryConfig>& output_mem_config = std::nullopt);
+Tensor _argmin(const Tensor& input_t, int64_t _dim, bool all, const std::optional<MemoryConfig>& output_mem_config = std::nullopt);
+
+struct ExecuteArgMax {
+    static Tensor operator()(
+        const Tensor& input_tensor,
+        int64_t dim,
+        bool all,
+        const std::optional<MemoryConfig>& memory_config = std::nullopt) {
+        return _argmax(input_tensor, dim, all, memory_config.value_or(input_tensor.memory_config()));
+    }
+};
```

Comment:
Misisng overload with `queue` and without `queue_id`

Reply:
its part of different milestone which we will start from Monday. As of now we only migrated the ops into ttnn.